lib/l10n/l10n.dart

```diff
@@ -127,6 +128,35 @@ class Language {
   String toString() => locale.toLanguageTag();
 }
 
+String formattedFileSize(int? bytes) {
+  if (bytes == null) return 'dot'.l10n * 3;
+
+  const precision = 1;
+  if (bytes < 1024) {
+    return 'label_b'.l10nfmt({'amount': bytes});
+  } else if (bytes < pow(1024, 2)) {
+    return 'label_kb'.l10nfmt({
+      'amount': (bytes / 1024).toPrecision(precision),
+    });
+  } else if (bytes < pow(1024, 3)) {
+    return 'label_mb'.l10nfmt({
+      'amount': (bytes / pow(1024, 2)).toPrecision(precision),
+    });
+  } else if (bytes < pow(1024, 4)) {
+    return 'label_gb'.l10nfmt({
+      'amount': (bytes / pow(1024, 3)).toPrecision(precision),
+    });
+  } else if (bytes < pow(1024, 5)) {
+    return 'label_tb'.l10nfmt({
+      'amount': (bytes / pow(1024, 4)).toPrecision(precision),
+    });
+  } else {
+    return 'label_pb'.l10nfmt({
+      'amount': (bytes / pow(1024, 5)).toPrecision(precision),
+    });
+  }
```

Comment:
I see a pattern here that might be probably rewritten into a code with lesser duplication? And can't bytes be compared with some bitwise operations to omit floating numbers computations?

Reply:
Yes, there is a pattern, but it's hard to eliminate reasonably.

One thing you could do is use some math:
```dart
String formattedFileSize(int? bytes) {
  if (bytes == null) return 'dot'.l10n * 3;
  if (bytes == 0) return 'label_b'.l10nfmt({'amount': '0'});

  const labels = [
    'label_b',
    'label_kb',
    'label_mb',
    'label_gb',
    'label_tb',
    'label_pb',
  ];

  final powerLevel = min(
    (log(bytes) / log(1024)).toInt(), // logₖ(x) = log(x) / log(k)
    labels.length - 1,
  );

  final value = bytes / pow(1024, powerLevel);
  final string = value.toStringAsFixed(1);

  return labels[powerLevel].l10nfmt({
    'amount': string.endsWith('.0') //
        ? string.substring(0, string.length - 2)
        : string,
  });
}
```

but I don't see how is it better than the previous version. It's roughly the same length, but it has a higher complexity.

I don't see any obvious bitwise algorithm.

Reply:
@Gieted, ok, I do agree, the `if`s approach is easier to understand. This overcomplicating won't help.